motor/motor_control/motorcontrol/motorcontrol.ino

```diff
@@ -0,0 +1,140 @@
+
+#include <ros.h>
+#include <ArduinoHardware.h>
+
+
+#include <ros.h>
+#include <std_msgs/Float64.h>
+#include <std_msgs/Bool.h>
+#include <tf/transform_broadcaster.h>
+#include <nav_msgs/Odometry.h>
+
+#include <ros/time.h>
+
+static unsigned long last_ros_count = 0;
+
+// ports (arbitrary numbers as placeholders)
+// right
+const int RFR =  4;      // digital; motor direction control; F/R == COM == 0 => anticlockwise; clockwise defined as forwards
+const int RSV = 5;       // controls speed; 5V = max speed
+// left
+const int LFR = 7;       // digital; motor direction control; F/R == COM == 0 => anticlockwise; clockwise defined as forwards
+const int LSV = 6;       // controls speed; 5V = max speed
+// commands
+const int light_pin = 13;  
+
+unsigned long current_ros = 0;
+
+
+// code for light (flashes when rover in autonomous mode, solid when manual)
+// light (code based on https://www.arduino.cc/en/Tutorial/BuiltInExamples/BlinkWithoutDelay)
+boolean mode = 1;     // 1 -> autonomous mode (light flashes), 0 -> manual mode (light solid)
+            // digital; placeholder pin number
+int light_state = LOW;                // used to set light
```

Comment:
You have a bunch of global variables.

I'd recommend prefixing the non constants with `g_`. For example `g_light_state` to make it clear it's global.

Additionally, make constants fully capitialized (e.g.  `INTERVAL`).

Reply:
technically speaking, aren't global constants marked with c_ while local ones are CAPITALIZED?